ads/google/test/test-doubleclick.js

```diff
@@ -0,0 +1,83 @@
+/**
+ * Copyright 2017 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {selectGptExperiment, writeAdScript} from '../doubleclick';
+
+describe('selectGptExperiment', () => {
+
+  it('should return the correct filename when given the experimental condition',
+  () => {
+    const controlData = {experimentId: '21060540'};
+    const experimentData = {experimentId: '21060541'};
+    const notInEitherData = {};
+    expect(selectGptExperiment(controlData)).to
+    .equal('gpt_sf_a.js');
+    expect(selectGptExperiment(experimentData)).to
+    .equal('gpt_sf_b.js');
+    expect(selectGptExperiment(notInEitherData)).to
+    .equal(undefined);
+  });
+});
+
+describe('writeAdScript', () => {
```

Comment:
These tests modify document and so are not hermetic causing script additions to be carried between tests!  You need to change to load within an iframe and then use its window in your tests (looking at win.document for query selectors):

let win;
beforeEach(() => {
   return createIframePromise().then(iframe => {
        win = iframe.contentWindow;
   });
});